### ISSUE TITLE
multiple calls to wiringPiSetup causes crash

### ISSUE THREAD

**Issue by Clotoo** [sesh-kebab/raspi-sonar#6]
Hello,

I am not sure what is missing for this to be included in raspi-io. You might want to bump up your thread there ?

In the meantime, I'm using this as a standalone module alongside johnny-five, and faced the same issue as described there (another standalone module) : https://github.com/dangrie158/opencushion/issues/2

A simple enough way to fix it is to let devs decide whether wiringPiSetup should be called or not, by adding a boolean argument to the constructor. It's not a pretty solution, but since wiringPi doesn't provide a way to know whether it's already initialized or not, it seems to be the only viable solution ATM.

Thanks

**Comment by sesh-kebab** [comment]
Hi there, thanks for letting me know. As for what's missing, nothing really. I had a working PR and fixed a lot of the issues. Problem was I was taking too long and I needed a rebase everytime I pushed my local changes which resulted in merge conflicts that were time consuming to get around. So I abandoned it as I was able to use this as a standalone module.

I'm going to have another crack at submitting a PR this week as it sounds like this module is getting some use 😊

As for the suggested workaround, seems like a reasonable workaround. Will go down that route if we can't get this into raspi-io this week.
